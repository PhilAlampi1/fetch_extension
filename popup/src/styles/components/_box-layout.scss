.box-layout {
    background: url('/images/buildings.jpeg');
    background-size: cover;
    display: flex;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(black, 0.25);
    border-radius: 3px;
    text-align: center;
    width: 25rem;
    margin: auto;
}

.box-layout__title {
    color: white;
    font-size: $xl-size;
    font-weight: bold;
}

.box-layout__subtext {
    color: white;
    font-size: $m-size;
    font-weight: 500;
    padding: 0 0 $s-size 0;
    text-align: center;
}

.box-layout__footer {
    display: flex;
    justify-content: space-around;
}

.box-layout__footertext {
    color: white;
    font-size: $s-size;
    margin: $s-size;
}

.box-layout__disclaimer {
    margin: $xl-size 0 0 0;
}