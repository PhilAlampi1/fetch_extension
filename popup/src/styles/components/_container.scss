.container {
    margin: 0 auto;
    max-width: 60rem;
    padding: 0 $m-size;
    overflow: scroll;
}

.container__centertext {
    text-align: center;
    margin: $xl-size 0 0 0;
    overflow: scroll;
}

.container__lefttext {
    text-align: left;
    margin: 0;
    overflow: scroll;
}

.container__flex {
    display: flex;
    justify-content: space-around;
    overflow: scroll;
}

.container--center {
    display: flex;
    align-content: center;
    justify-content: center;
    overflow: scroll;
}

.container__flexvertical {
    display: flex;
    flex-direction: column;
  }

.container__addspace {
    margin: $s-size 0;
}

.container__centerify {
    display: flex;
    align-items: center;
    text-align: center;
    margin: $xl-size 0 0 0;
    overflow: scroll;
}

.container__child {
    flex: 1;
    overflow: scroll;
}

.add-space {
    margin: $xl-size 0 0 0;
}