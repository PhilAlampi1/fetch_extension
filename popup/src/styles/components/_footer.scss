.footer {
    display: flex;
    justify-content: space-around;
}

.footer-item {
    color: black;
    font-size: $m-size;
    margin: $xl-size $s-size 1rem $s-size;
}