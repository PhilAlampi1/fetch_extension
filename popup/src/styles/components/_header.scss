.header {
    align-items: flex-start;
    background: $off-black;
    color: white;
    display: flex;
    padding: $m-size 0;
}

.header__container {
    align-items: center;
    display: flex;
    flex: 1;
    justify-content: space-between;
    padding: 0 $m-size;
}

.header__title {
    font-size: $l-size;
    margin: 0 $m-size;
    padding: $s-size;
    text-align: left;
    flex: 1;
}

.header__linkgroup {
    flex: 4;
}

.header__links {
    color: $off-white;
    font-size: $m-size;
    font-weight: 500;
    margin: $s-size;
    padding: $s-size;
    text-align: center;
}

.header__span {
    width: 100%
}

.header-question-parent {
    align-items: center;
    display: flex;
    justify-content: flex-start;
}

.header-question-child {
    font-size: $m-size;
    margin: 0 $s-size;
    padding: $s-size;
    width: 100px;
}