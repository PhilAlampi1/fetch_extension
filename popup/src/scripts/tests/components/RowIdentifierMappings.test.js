import React from 'react'
import { shallow } from 'enzyme'
import { RowIdentifierMappings } from '../../components/RowIdentifierMappings'
import {
    rowIdentifiers,
    importedData,
    standardFields,
    importRowIdentifierValues,
    importedRowIdentifierField
} from '../../../../../event/src/tests/fixtures/imports'

const importedRowIdentifierValue = "2983 Briggs Avenue"
const rowIdentifier = rowIdentifiers[0]
const setImportedRowIdentifierValue = "2731 Kingsbridge Terrace"
const rowIdentifiersObj = rowIdentifiers

test('RowIdentifierMappings should render correctly', () => {
    const wrapper = shallow(<RowIdentifierMappings
        rowIdentifiers={rowIdentifiersObj}
        importedData
        importedRowIdentifierField
        setImportedRowIdentifierValue
        importRowIdentifierValues
    />)
    expect(wrapper).toMatchSnapshot()
})

test('RowIdentifierMappings should call confirmRowIdentifiers when button is clicked', () => {
    const onClickSpy = jest.fn()
    const wrapper = shallow(<RowIdentifierMappings
        rowIdentifiers={rowIdentifiersObj}
        importedData
        importedRowIdentifierField
        setImportedRowIdentifierValue
        importRowIdentifierValues
        confirmRowIdentifiers = {onClickSpy}
    />)
    //LEFT OFF HERE - GETTING ERROR ON TOHAVEBEENLASTCALLEDWITH
    wrapper.find('button').simulate('click')
    expect(onClickSpy).toHaveBeenCalled()
})