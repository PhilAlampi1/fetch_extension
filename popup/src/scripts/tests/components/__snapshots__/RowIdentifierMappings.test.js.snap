// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowIdentifierMappings should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RowIdentifierMappings
    importRowIdentifierValues={true}
    importedData={true}
    importedRowIdentifierField={true}
    rowIdentifiers={
      Array [
        Object {
          "rowIdentifierId": "1",
          "rowIdentifierName": "Listing Comp 1",
          "rowIdentifierPrefix": "L1",
        },
        Object {
          "rowIdentifierId": "2",
          "rowIdentifierName": "Listing Comp 2",
          "rowIdentifierPrefix": "L2",
        },
        Object {
          "rowIdentifierId": "3",
          "rowIdentifierName": "Listing Comp 3",
          "rowIdentifierPrefix": "L3",
        },
      ]
    }
    setImportedRowIdentifierValue={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Which rows would you like to import?
        </p>,
        <div>
          <div />
          <div>
            Type
          </div>
          <div>
            Prefix
          </div>
        </div>,
        Array [
          <RowIdentifierLineItem
            importRowIdentifierValues={true}
            importedRowIdentifierValue={undefined}
            rowIdentifier={
              Object {
                "rowIdentifierId": "1",
                "rowIdentifierName": "Listing Comp 1",
                "rowIdentifierPrefix": "L1",
              }
            }
            rowIdentifierId="1"
            rowIdentifiers={
              Array [
                Object {
                  "rowIdentifierId": "1",
                  "rowIdentifierName": "Listing Comp 1",
                  "rowIdentifierPrefix": "L1",
                },
                Object {
                  "rowIdentifierId": "2",
                  "rowIdentifierName": "Listing Comp 2",
                  "rowIdentifierPrefix": "L2",
                },
                Object {
                  "rowIdentifierId": "3",
                  "rowIdentifierName": "Listing Comp 3",
                  "rowIdentifierPrefix": "L3",
                },
              ]
            }
            setImportedRowIdentifierValue={true}
          />,
          <RowIdentifierLineItem
            importRowIdentifierValues={true}
            importedRowIdentifierValue={undefined}
            rowIdentifier={
              Object {
                "rowIdentifierId": "2",
                "rowIdentifierName": "Listing Comp 2",
                "rowIdentifierPrefix": "L2",
              }
            }
            rowIdentifierId="2"
            rowIdentifiers={
              Array [
                Object {
                  "rowIdentifierId": "1",
                  "rowIdentifierName": "Listing Comp 1",
                  "rowIdentifierPrefix": "L1",
                },
                Object {
                  "rowIdentifierId": "2",
                  "rowIdentifierName": "Listing Comp 2",
                  "rowIdentifierPrefix": "L2",
                },
                Object {
                  "rowIdentifierId": "3",
                  "rowIdentifierName": "Listing Comp 3",
                  "rowIdentifierPrefix": "L3",
                },
              ]
            }
            setImportedRowIdentifierValue={true}
          />,
          <RowIdentifierLineItem
            importRowIdentifierValues={true}
            importedRowIdentifierValue={undefined}
            rowIdentifier={
              Object {
                "rowIdentifierId": "3",
                "rowIdentifierName": "Listing Comp 3",
                "rowIdentifierPrefix": "L3",
              }
            }
            rowIdentifierId="3"
            rowIdentifiers={
              Array [
                Object {
                  "rowIdentifierId": "1",
                  "rowIdentifierName": "Listing Comp 1",
                  "rowIdentifierPrefix": "L1",
                },
                Object {
                  "rowIdentifierId": "2",
                  "rowIdentifierName": "Listing Comp 2",
                  "rowIdentifierPrefix": "L2",
                },
                Object {
                  "rowIdentifierId": "3",
                  "rowIdentifierName": "Listing Comp 3",
                  "rowIdentifierPrefix": "L3",
                },
              ]
            }
            setImportedRowIdentifierValue={true}
          />,
        ],
        <button
          onClick={[Function]}
        >
          Next
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Which rows would you like to import?",
        },
        "ref": null,
        "rendered": "Which rows would you like to import?",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div />,
            <div>
              Type
            </div>,
            <div>
              Prefix
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Type",
            },
            "ref": null,
            "rendered": "Type",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Prefix",
            },
            "ref": null,
            "rendered": "Prefix",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "importRowIdentifierValues": true,
          "importedRowIdentifierValue": undefined,
          "rowIdentifier": Object {
            "rowIdentifierId": "1",
            "rowIdentifierName": "Listing Comp 1",
            "rowIdentifierPrefix": "L1",
          },
          "rowIdentifierId": "1",
          "rowIdentifiers": Array [
            Object {
              "rowIdentifierId": "1",
              "rowIdentifierName": "Listing Comp 1",
              "rowIdentifierPrefix": "L1",
            },
            Object {
              "rowIdentifierId": "2",
              "rowIdentifierName": "Listing Comp 2",
              "rowIdentifierPrefix": "L2",
            },
            Object {
              "rowIdentifierId": "3",
              "rowIdentifierName": "Listing Comp 3",
              "rowIdentifierPrefix": "L3",
            },
          ],
          "setImportedRowIdentifierValue": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "importRowIdentifierValues": true,
          "importedRowIdentifierValue": undefined,
          "rowIdentifier": Object {
            "rowIdentifierId": "2",
            "rowIdentifierName": "Listing Comp 2",
            "rowIdentifierPrefix": "L2",
          },
          "rowIdentifierId": "2",
          "rowIdentifiers": Array [
            Object {
              "rowIdentifierId": "1",
              "rowIdentifierName": "Listing Comp 1",
              "rowIdentifierPrefix": "L1",
            },
            Object {
              "rowIdentifierId": "2",
              "rowIdentifierName": "Listing Comp 2",
              "rowIdentifierPrefix": "L2",
            },
            Object {
              "rowIdentifierId": "3",
              "rowIdentifierName": "Listing Comp 3",
              "rowIdentifierPrefix": "L3",
            },
          ],
          "setImportedRowIdentifierValue": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "importRowIdentifierValues": true,
          "importedRowIdentifierValue": undefined,
          "rowIdentifier": Object {
            "rowIdentifierId": "3",
            "rowIdentifierName": "Listing Comp 3",
            "rowIdentifierPrefix": "L3",
          },
          "rowIdentifierId": "3",
          "rowIdentifiers": Array [
            Object {
              "rowIdentifierId": "1",
              "rowIdentifierName": "Listing Comp 1",
              "rowIdentifierPrefix": "L1",
            },
            Object {
              "rowIdentifierId": "2",
              "rowIdentifierName": "Listing Comp 2",
              "rowIdentifierPrefix": "L2",
            },
            Object {
              "rowIdentifierId": "3",
              "rowIdentifierName": "Listing Comp 3",
              "rowIdentifierPrefix": "L3",
            },
          ],
          "setImportedRowIdentifierValue": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Next",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Next",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Which rows would you like to import?
          </p>,
          <div>
            <div />
            <div>
              Type
            </div>
            <div>
              Prefix
            </div>
          </div>,
          Array [
            <RowIdentifierLineItem
              importRowIdentifierValues={true}
              importedRowIdentifierValue={undefined}
              rowIdentifier={
                Object {
                  "rowIdentifierId": "1",
                  "rowIdentifierName": "Listing Comp 1",
                  "rowIdentifierPrefix": "L1",
                }
              }
              rowIdentifierId="1"
              rowIdentifiers={
                Array [
                  Object {
                    "rowIdentifierId": "1",
                    "rowIdentifierName": "Listing Comp 1",
                    "rowIdentifierPrefix": "L1",
                  },
                  Object {
                    "rowIdentifierId": "2",
                    "rowIdentifierName": "Listing Comp 2",
                    "rowIdentifierPrefix": "L2",
                  },
                  Object {
                    "rowIdentifierId": "3",
                    "rowIdentifierName": "Listing Comp 3",
                    "rowIdentifierPrefix": "L3",
                  },
                ]
              }
              setImportedRowIdentifierValue={true}
            />,
            <RowIdentifierLineItem
              importRowIdentifierValues={true}
              importedRowIdentifierValue={undefined}
              rowIdentifier={
                Object {
                  "rowIdentifierId": "2",
                  "rowIdentifierName": "Listing Comp 2",
                  "rowIdentifierPrefix": "L2",
                }
              }
              rowIdentifierId="2"
              rowIdentifiers={
                Array [
                  Object {
                    "rowIdentifierId": "1",
                    "rowIdentifierName": "Listing Comp 1",
                    "rowIdentifierPrefix": "L1",
                  },
                  Object {
                    "rowIdentifierId": "2",
                    "rowIdentifierName": "Listing Comp 2",
                    "rowIdentifierPrefix": "L2",
                  },
                  Object {
                    "rowIdentifierId": "3",
                    "rowIdentifierName": "Listing Comp 3",
                    "rowIdentifierPrefix": "L3",
                  },
                ]
              }
              setImportedRowIdentifierValue={true}
            />,
            <RowIdentifierLineItem
              importRowIdentifierValues={true}
              importedRowIdentifierValue={undefined}
              rowIdentifier={
                Object {
                  "rowIdentifierId": "3",
                  "rowIdentifierName": "Listing Comp 3",
                  "rowIdentifierPrefix": "L3",
                }
              }
              rowIdentifierId="3"
              rowIdentifiers={
                Array [
                  Object {
                    "rowIdentifierId": "1",
                    "rowIdentifierName": "Listing Comp 1",
                    "rowIdentifierPrefix": "L1",
                  },
                  Object {
                    "rowIdentifierId": "2",
                    "rowIdentifierName": "Listing Comp 2",
                    "rowIdentifierPrefix": "L2",
                  },
                  Object {
                    "rowIdentifierId": "3",
                    "rowIdentifierName": "Listing Comp 3",
                    "rowIdentifierPrefix": "L3",
                  },
                ]
              }
              setImportedRowIdentifierValue={true}
            />,
          ],
          <button
            onClick={[Function]}
          >
            Next
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Which rows would you like to import?",
          },
          "ref": null,
          "rendered": "Which rows would you like to import?",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div />,
              <div>
                Type
              </div>,
              <div>
                Prefix
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Type",
              },
              "ref": null,
              "rendered": "Type",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Prefix",
              },
              "ref": null,
              "rendered": "Prefix",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "importRowIdentifierValues": true,
            "importedRowIdentifierValue": undefined,
            "rowIdentifier": Object {
              "rowIdentifierId": "1",
              "rowIdentifierName": "Listing Comp 1",
              "rowIdentifierPrefix": "L1",
            },
            "rowIdentifierId": "1",
            "rowIdentifiers": Array [
              Object {
                "rowIdentifierId": "1",
                "rowIdentifierName": "Listing Comp 1",
                "rowIdentifierPrefix": "L1",
              },
              Object {
                "rowIdentifierId": "2",
                "rowIdentifierName": "Listing Comp 2",
                "rowIdentifierPrefix": "L2",
              },
              Object {
                "rowIdentifierId": "3",
                "rowIdentifierName": "Listing Comp 3",
                "rowIdentifierPrefix": "L3",
              },
            ],
            "setImportedRowIdentifierValue": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "importRowIdentifierValues": true,
            "importedRowIdentifierValue": undefined,
            "rowIdentifier": Object {
              "rowIdentifierId": "2",
              "rowIdentifierName": "Listing Comp 2",
              "rowIdentifierPrefix": "L2",
            },
            "rowIdentifierId": "2",
            "rowIdentifiers": Array [
              Object {
                "rowIdentifierId": "1",
                "rowIdentifierName": "Listing Comp 1",
                "rowIdentifierPrefix": "L1",
              },
              Object {
                "rowIdentifierId": "2",
                "rowIdentifierName": "Listing Comp 2",
                "rowIdentifierPrefix": "L2",
              },
              Object {
                "rowIdentifierId": "3",
                "rowIdentifierName": "Listing Comp 3",
                "rowIdentifierPrefix": "L3",
              },
            ],
            "setImportedRowIdentifierValue": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "importRowIdentifierValues": true,
            "importedRowIdentifierValue": undefined,
            "rowIdentifier": Object {
              "rowIdentifierId": "3",
              "rowIdentifierName": "Listing Comp 3",
              "rowIdentifierPrefix": "L3",
            },
            "rowIdentifierId": "3",
            "rowIdentifiers": Array [
              Object {
                "rowIdentifierId": "1",
                "rowIdentifierName": "Listing Comp 1",
                "rowIdentifierPrefix": "L1",
              },
              Object {
                "rowIdentifierId": "2",
                "rowIdentifierName": "Listing Comp 2",
                "rowIdentifierPrefix": "L2",
              },
              Object {
                "rowIdentifierId": "3",
                "rowIdentifierName": "Listing Comp 3",
                "rowIdentifierPrefix": "L3",
              },
            ],
            "setImportedRowIdentifierValue": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Next",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Next",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
